### ISSUE TITLE
`opening directory $(pwd)/.direnv: No such file or directory`

### ISSUE THREAD

**Issue by RasmusRendal** [nix-community/nix-direnv#180]
I get this bug every time I try to use nix-direnv in a directory with a flake:
```bash
rendal@machine  ~/Projects/utap# echo "use flake ." >> .envrc
direnv: error /home/rendal/Projects/utap/.envrc is blocked. Run `direnv allow` to approve its content                                                                                                                                                                                                                       
➜ rendal@129857  ~/Projects/utap# direnv allow
direnv: loading ~/Projects/utap/.envrc                                                                                                                                                                                                                                                                                      
direnv: using flake .
error: opening directory '/home/rendal/Projects/utap/.direnv': No such file or directory
```

I can also confirm that there really is no such directory.
If I simply create the directory myself, the problem goes away.

**Comment by bbenne10** [comment]
A few followup questions:

* What version of nix-direnv?
* How are you setting up nix-direnv to run? If you're using home-manager or configuration.nix, please let me see the relevant configuration sections.
* What does your project's `.envrc` look like?
* What does `~/.config/direnv/direnvrc` look like (if it exists)?

**Comment by RasmusRendal** [comment]
> A few followup questions:
> 
>     * What version of nix-direnv?
> 
>     * How are you setting up nix-direnv to run? If you're using home-manager or configuration.nix, please let me see the relevant configuration sections.
> 
>     * What does your project's `.envrc` look like?
> 
>     * What does `~/.config/direnv/direnvrc` look like (if it exists)?

Thanks you for responding! :D

## What version of nix-direnv?
Nix-direnv is 1.4.0
## How are you setting up nix-direnv to run? If you're using home-manager or configuration.nix, please let me see the relevant configuration sections.
I'm using configuration.nix to set up nix-direnv:
```nix
{
nixpkgs.overlays = [
    (self: super: { nix-direnv = super.nix-direnv.override { enableFlakes = true; }; })
  ];

programs.zsh = {
    enable = true;
    ohMyZsh = {
      enable = true;
      plugins = [ "git" "ssh-agent" "direnv" ];
      theme = "kafeitu";
    };
  };

environment.systemPackages = with pkgs; [ nix-direnv others ];
}
```

## What does your project's `.envrc` look like?
```
use flake .
```
## What does ~/.config/direnv/direnvrc look like (if it exists)?
It does not exist, and there is not a directory there either.

**Comment by bbenne10** [comment]
`1.4.0` is both the initial implementation of `use_flake` and from June of last year. Is there some reason you cannot update to the current `2.1.0` release? I don't see significant differences in between the two implementations, but I am testing it out now. 

If there's a strong reason to stay with `1.4.0` and this is a problem with `nix-direnv` (and not your setup surrounding it), I might recommend forking and using that fork rather than nixpkg's bundled version. Personally, I don't have the bandwidth to support old releases concurrently with new ones at the moment and so `1.4.0` is not going to see new commits from me.


Update: On macOS 10.15, with a nix-provided `bash`, I cannot replicate this.
My test procedure was to checkout the `1.4.0` tag of this repository, and modify `tests/testenv/.envrc` to change `use nix` to `use flake` (since the directory contains both a `shell.nix` and a `flake.nix`). I also modified the `direnvrc` it sources to add `set -x` at the top of `use_flake`. (You may also want to add a `set +x` at the bottom of the function so that we're not debugging the direnv internals :) )

Would you be able to perform the same procedure and post the output? `set -x` should give us an exact line to look at and then we can see if there's some more poking we need to do

**Comment by bbenne10** [comment]
Ping? If there's no update, I'm going to close this as solved by updating to 2.x.

**Comment by RasmusRendal** [comment]
Hey, sorry for the late reply. I did successfully reproduce the issue:
```
$ direnv allow
direnv: loading /tmp/nix-direnv/.envrc                                                                                                                                                                                                      
direnv: using flake
[1/0/1 copied (110.5/110.5 MiB), 19.1 MiB DL] fetching source from https://cache.nixos.orgdirenv: ([direnv export zsh]) is taking a while to execute. Use CTRL-C to give up.
error: opening directory '/tmp/nix-direnv/.direnv': No such file or directory
```

The reason that I was using 1.40 is simply that was the one in the latest stable Nixpkgs. However, the issue still happened with 2.1.0. It was fixed when I updated direnv itself from 2.29 to 2.31, which seems to indicate it wasn't even your issue. I don't know how much you use direnv's stdlib, but maybe the following PR did it: https://github.com/direnv/direnv/pull/861

In any case, updating direnv fixes the issue. Thanks for your help!
